clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/client/TitusClientAuthenticationUtil.java

```diff
@@ -33,7 +35,21 @@
   public static <STUB extends AbstractStub<STUB>> STUB attachCaller(STUB serviceStub) {
     Metadata metadata = new Metadata();
     metadata.put(CALLER_ID_KEY, AuthenticatedRequest.getSpinnakerUser().orElse("spinnaker"));
-    metadata.put(CALL_REASON_KEY, AuthenticatedRequest.getSpinnakerExecutionId().orElse("unknown"));
+    metadata.put(
+        CALL_REASON_KEY,
+        String.format(
+            "Invoked by Spinnaker execution %s, Application name %s",
+            getSpinnakerExecutionId(
+                AuthenticatedRequest.getSpinnakerExecutionId().orElse("unknown")),
+            MDC.get(AuthenticatedRequest.Header.APPLICATION.getHeader())));
     return serviceStub.withInterceptors(MetadataUtils.newAttachHeadersInterceptor(metadata));
   }
+
+  public static String getSpinnakerExecutionId(String executionIdHeader) {
+    if (!executionIdHeader.equalsIgnoreCase("unknown") && !executionIdHeader.isEmpty()) {
```

Comment:
```suggestion
    if (!"unknown".equalsIgnoreCase(executionIdHeader) && !executionIdHeader.isEmpty()) {
```

Protect against NPEs! Also, would recommend adding `@Nonnull` or `@Nullable` on public method return types & arguments.